tests/.coveragerc

```diff
@@ -1,16 +1,43 @@
 [run]
 branch = True
 concurrency = multiprocessing,thread
-data_file = ${RUNTESTS_DIR-.}/.coverages/.coverage
+data_file = ${RUNTESTS_DIR-.}/.coverage
 omit =
     */django/utils/autoreload.py
-source = django
+    */tests/*
+    */migrations/*
+    */site-packages/*
+    */venv/*
+    */env/*
+    */__pycache__/*
+    */build/*
+    */dist/*
+    */django/conf/locale/*
+source = ${RUNTESTS_DIR-.}/django
 
 [report]
 ignore_errors = True
 omit =
     */django/conf/locale/*
     */tests/*
+    */migrations/*
+    */site-packages/*
+    */venv/*
+    */env/*
+    */__pycache__/*
+    */build/*
+    */dist/*
+exclude_lines =
+    pragma: no cover
+    def __repr__
+    if self.debug:
+    if settings.DEBUG
+    raise AssertionError
+    raise NotImplementedError
+    if 0:
+    if __name__ == .__main__.:
+    class .*\bProtocol\):
+    @(abc\.)?abstractmethod
```

Comment:
I think each of these changes need an explanation

Reply:
None are strictly mandatory for coverage to work; they just reduce noise and keep the signal focused on executable, testable paths. 
I can tighten or relax these based on how strict you want diff coverage to be.
e.g. - “exclude_lines” (lines 30–40) keeps files measured but ignores specific line patterns in reporting.
let me know if you have something else in mind.

Reply:
I think with a lot of these, I would still expect test coverage